### PR TITLE
PS-5695: Reformat scrubbing code (8.0)

### PR DESCRIPTION
**Pull request by dutow**
(cherry picked from commit 7cb01a7045725553f0043c5adc9ea21ebb18fa46)

### CODE REVIEW

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
Same comments as for 5.7

File: storage/innobase/btr/btr0scrub.cc [hunk 1]
```diff
@@ -462,9 +442,10 @@ static dberr_t btr_pessimistic_scrub(
...
 
-/****************************************************************
-Location index by id for a table
-return index or NULL */
+/** Location index by id for a table
+@param[in]	table
+@param[in]	index_id
+@return		index or NULL  */
 static dict_index_t *find_index(
     /*========*/
```
Comment (laurynas-biveinis) [inline storage/innobase/btr/btr0scrub.cc]:
```suggestion
```

File: storage/innobase/include/btr0scrub.h [hunk 2]
```diff
@@ -28,13 +28,12 @@ enum btr_scrub_page_allocation_status_t {
 #define BTR_SCRUB_SKIP_PAGE_AND_CLOSE_TABLE 3    /* no scrub & close table */
 #define BTR_SCRUB_SKIP_PAGE_AND_COMPLETE_SPACE 4 /* no scrub & complete space \
                                                   */
-#define BTR_SCRUB_TURNED_OFF      \
-  5 /* we detected that scrubbing \
-    was disabled by global        \
-    variable */
+#define BTR_SCRUB_TURNED_OFF                                      \
+  5 /* we detected that scrubbing                                 \
```
Comment (laurynas-biveinis) [inline storage/innobase/include/btr0scrub.h]: Needs manual help: join the comment lines and then reformat

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: storage/innobase/include/btr0scrub.h (hunk 2 above)
Comment (dutow) [reply storage/innobase/include/btr0scrub.h]: There's something bugged here: repeated clang format calls keep changing the formatting... looks like it stabilizes after a few calls, I'll commit that version.

File: storage/innobase/btr/btr0scrub.cc [hunk 3]
```diff
@@ -779,31 +739,26 @@ static void btr_scrub_update_total_stat(btr_scrub_t *scrub_data) {
...
-
 void btr_scrub_complete_space(btr_scrub_t *scrub_data) {
   ut_ad(scrub_data->scrubbing);
   btr_scrub_table_close_for_thread(scrub_data);
   btr_scrub_update_total_stat(scrub_data);
 }
 
-/*********************************************************************
-Return scrub statistics */
+/* Return scrub statistics
```
Comment (laurynas-biveinis) [inline storage/innobase/btr/btr0scrub.cc]:
```suggestion
/** Return scrub statistics
```

File: storage/innobase/include/btr0scrub.h [hunk 4]
```diff
@@ -23,18 +23,16 @@ enum btr_scrub_page_allocation_status_t {
...
-    variable */
-
-/**************************************************************/ /**
- struct for keeping scrub statistics. */
+#define BTR_SCRUB_PAGE 1                      /* page should be scrubbed */
+#define BTR_SCRUB_SKIP_PAGE 2                 /* no scrub & no action */
+#define BTR_SCRUB_SKIP_PAGE_AND_CLOSE_TABLE 3 /* no scrub & close table */
+#define BTR_SCRUB_SKIP_PAGE_AND_COMPLETE_SPACE \
+  4 /* no scrub & complete space               \
+     */
```
Comment (laurynas-biveinis) [inline storage/innobase/include/btr0scrub.h]: No need for a new line to close a comment? Consider // if it helps

File: storage/innobase/include/btr0scrub.h [hunk 5]
```diff
@@ -68,75 +65,62 @@ struct btr_scrub_t {
...
-    btr_scrub_t *scrub_data,                      /*!< inut: scrub data */
-    buf_block_t *block,                           /*!< in: block */
-    btr_scrub_page_allocation_status_t allocated, /*!< in: is block
-                                                  allocated or free */
-    mtr_t *mtr);                                  /*!< in: mtr */
-
-/****************************************************************
-Perform actual scrubbing of page */
 int btr_scrub_page(
     /*============*/
```
Comment (laurynas-biveinis) [inline storage/innobase/include/btr0scrub.h]:
```suggestion
```

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by dutow** [comment]
Updated. https://ps80.cd.percona.com/job/percona-server-8.0-param/217/

File: storage/innobase/btr/btr0scrub.cc [hunk 6]
```diff
@@ -268,7 +258,7 @@ int btr_page_needs_scrubbing(
          */
 
         /* if encountering page type not needing scrubbing
-        release reference to table object */
+release reference to table object */
```
Comment (laurynas-biveinis) [inline storage/innobase/btr/btr0scrub.cc]: Oh no, why? Can this be avoided?

Comment (dutow) [reply storage/innobase/btr/btr0scrub.cc]: I'm not sure what happened there... I changed it back and now clang-format leaves it alone.

**Review by laurynas-biveinis — APPROVED** [review]
OK once CircleCI OK

**Pull request merged**